machine/v1/types_alibabaprovider.go

```diff
@@ -120,9 +134,11 @@ type AlibabaCloudMachineProviderConfig struct {
 	// +optional
 	RAMRoleName string `json:"ramRoleName,omitempty"`
 
-	// ResourceGroupID is the unique ID of the resource group to which to assign the instance.
+	// ResourceGroup references the resource group to which to assign the instance.
+	// A reference holds either the resource group ID, the resource name, or the required tags to search.
```

Comment:
Resource groups cannot use tags to search, we should update this comment to note that tags can't be used with a resource group lookup

Reply:
i'll add that